### PR TITLE
fix: Tizen video error fixed by checking the extended MIME type

### PR DESCRIPTION
**Pull request by FernandoGarciaDiez**
We have reproduced an issue in some tizen tv's, and we think it's the cause of #4634 and possibly #4503 

It happens with some streams that have 3 variants, 2 of them are supported on these TVs, but the one with higher resolution isn't. This variant uses this video: video/mp4;codecs="avc1.64002a";framerate="25";bitrate="6000000";width="**1980**";height="1080" and in this TV the higher resolution that is supported is **1920**x1080

At **filterManifestByMediaCapabilities** function from **stream_utils.js** we were checking `Capabilities.isTypeSupported(fullType)` , where this "fullType" only includes the mime type and the codecs (`video/mp4; codecs="avc1.64002a"` for example). This is currently returning true, so the player believes that this variant is supported, but then it fails when it tries to play it.

However, there was an older implementation that used **shaka.media.MediaSourceEngine.isStreamSupported()** function which internally checks `Capabilities.isTypeSupported(extendedMimeType)`, where "extendedMimeType" includes also the width, height, bitrate and framerate (for example, `video/mp4;codecs="avc1.64002a";framerate="25";bitrate="6000000";width="1980";height="1080"`), which in this specific case returns false because in this case the 1980 width is not supported (it should be 1920), the variant is filtered out, and the playback is started with the other 2 variants.

In this proposed change I'm basically changing isTypeSupported(fullType) by isTypeSupported(extendedMimeType), keeping in mind that the codecs and mimeType could have been modified in the call to getFullOrConvertedType and we must keep using those modified values.
In this change I haven't modified the mutiplexed streams case (the case from the `if (video.codecs.includes(','))` at filterManifestByMediaCapabilities), because the getExtendedType function gets the whole variant as parameter, I don't have any good way of testing the case, and also I think that the audio tracks are less likely to be affected by this issue, so it's probably not that important.

Fixes https://github.com/shaka-project/shaka-player/issues/4634

### CODE REVIEW

**Comment by avelad** [comment]
@FernandoGarciaDiez Please, sign the CLA. Thanks!

**Comment by FernandoGarciaDiez** [comment]
I did already, but I think that as it is a corporate CLA from a new corporation, it can take some days until it's processed (https://opensource.google/documentation/reference/cla#cla-accepted)

**Comment by joeyparrish** [comment]
I can see the logic of this, however, resolution constraints are meant to be checked through MediaCapabilities.

Is MediaCapabilities implemented on this model of Tizen, or is our polyfill being used instead?

If it's the polyfill, let's make this change at that level (building the extended MIME type there).

If it's a broken native implementation, let's still make the change at the level of the polyfill, but let's force its use on Tizen.

**Comment by FernandoGarciaDiez** [comment]
I see what you mean, yes, the polyfill is installed and I see how that's where the fix should be, as it is not taking into account the resolution. Another possible issue is that I am not 100% sure that even if the polyfill is installed it is working as expected, it looks as if it was never called, but as I say, I still need to investigate it a bit more

**Comment by FernandoGarciaDiez** [comment]
As I'm working on this, I have realized that this issue is pretty much the same as #4726 which was done exclusively for the Cast platform. @JulianDomingo since you worked on that issue some months ago, why was is so tailored for the video component just in the cast platform? Do you think there would be any problem in checking the extended mime type parameters in all platforms and both for video and audio? Or would it be better to add some similar code only for tizen video?

**Comment by joeyparrish** [comment]
> why was is so tailored for the video component just in the cast platform?

In general, these "extended MIME types" are non-standard, and the parameters are ignored by most platforms.  Cast is an exception, though I hope to see the platform embrace the web-standard way through the MediaCapabilities API.

> Only for Tizen, because there are some caches with https://github.com/shaka-project/shaka-player/blob/main/lib/polyfill/media_capabilities.js#L124, and the extended mime type produces many new possibilities

Also, this.  :grin:

File: lib/polyfill/media_capabilities.js [hunk 1]
```diff
@@ -119,6 +119,13 @@ shaka.polyfill.MediaCapabilities = class {
         if (shaka.util.Platform.isChromecast()) {
           isSupported =
               shaka.polyfill.MediaCapabilities.canCastDisplayType_(videoConfig);
+        } else if (shaka.util.Platform.isTizen()) {
+          let extendedType = videoConfig.contentType;
+          extendedType += `; width=${videoConfig.width}`;
```
Comment (avelad) [inline lib/polyfill/media_capabilities.js:124]: This 4 properties can be null in some streams, please check if the property is null before append. Thanks!

**Review by avelad — CHANGES_REQUESTED** [review]

File: lib/polyfill/media_capabilities.js (hunk 1 above)
Comment (FernandoGarciaDiez) [reply lib/polyfill/media_capabilities.js]: I thought these were all mandatory parameters, but I guess that doesn't mean they can't possibly be null in some type of streams.

**Comment by avelad** [comment]
I'll run the GitHub tests and the lab.

**Review by avelad — APPROVED** [review]

**Review by theodab — APPROVED** [review]

**Review by joeyparrish — APPROVED** [review]

**Pull request merged**